packages/sit-onyx/src/components/OnyxButton/OnyxButton.vue

```diff
@@ -0,0 +1,182 @@
+  --border-color: transparent;
+  --disabled-border-color: transparent;
+  --focus-outline-color: var(--onyx-color-base-primary-200, #bfddf3);
+
+  &--secondary {
+    --background-hover-color: var(--onyx-color-base-neutral-200, #eaedf0);
+    --focus-outline-color: var(--onyx-color-base-neutral-300, #d8dde1);
+    --color: var(--onyx-color-text-icons-neutral-intense, #52626d);
+    --disabled-color: var(--onyx-color-text-icons-primary-soft, #bfddf3);
+  }
+
+  &--danger {
+    --background-hover-color: var(--onyx-color-base-danger-200, #f1d3d2);
+    --focus-outline-color: var(--onyx-color-base-danger-300, #e6a7a6);
+    --color: var(--onyx-color-text-icons-danger-intense, #d1332f);
+    --disabled-color: var(--onyx-color-text-icons-danger-medium, #e6a7a6);
+  }
+
+  &--default--primary {
+    --background-color: var(--onyx-color-base-primary-500, #0079ce);
+    --background-hover-color: var(--onyx-color-base-primary-400, #3b98d9);
+    --background-disabled-color: var(--onyx-color-base-primary-200, #bfddf3);
+    --color: var(--onyx-color-text-icons-neutral-inverted, #fff);
+    --disabled-color: var(--onyx-color-text-icons-neutral-inverted, #fff);
+    --border-color: var(--onyx-color-base-primary-500, #0079ce);
+    --disabled-border-color: var(--onyx-color-base-primary-200, #bfddf3);
+  }
+
+  &--default--secondary {
+    --background-color: var(--onyx-color-base-background-blank, #fff);
+    --background-hover-color: var(--onyx-color-base-neutral-200, #eaedf0);
+    --background-disabled-color: var(--onyx-color-base-background-blank, #fff);
+    --border-color: var(--onyx-color-base-neutral-400, #b7bfc5);
+    --disabled-border-color: var(--onyx-color-base-neutral-200, #eaedf0);
+  }
+
+  &--default--danger {
+    --background-color: var(--onyx-color-base-danger-200, #f1d3d2);
+    --background-hover-color: var(--onyx-color-base-danger-100, #fbefee);
+    --background-disabled-color: var(--onyx-color-base-danger-100, #fbefee);
+    --border-color: var(--onyx-color-base-danger-500, #d1332f);
+    --disabled-border-color: var(--onyx-color-base-danger-200, #f1d3d2);
+  }
+
+  &--outline--primary {
+    --border-color: var(--onyx-color-base-primary-500, #0079ce);
+    --disabled-border-color: var(--onyx-color-base-primary-200, #bfddf3);
+  }
+
+  &--outline--secondary {
+    --border-color: var(--onyx-color-base-neutral-400, #b7bfc5);
+    --disabled-border-color: var(--onyx-color-base-neutral-200, #eaedf0);
+  }
+
+  &--outline--danger {
+    --border-color: var(--onyx-color-base-danger-500, #d1332f);
+    --disabled-border-color: var(--onyx-color-base-danger-200, #f1d3d2);
+  }
+
+  background-color: var(--background-color);
+  border: 0.063rem solid var(--border-color);
+
+  &:hover {
+    background: var(--background-hover-color);
+  }
+
+  &:focus-visible {
+    outline: 0.25rem solid var(--focus-outline-color);
+  }
+
+  &--disabled {
+    cursor: none;
+    pointer-events: none;
+    background-color: var(--background-disabled-color);
```

Comment:
You don't need an additional css property for the disabled state, you can just use the `--background-color` one

Reply:
For reference you can take a look at my `RadioButton` component implementation here: 
https://github.com/SchwarzIT/onyx/pull/275/files#diff-3bc1a3fa1286bfb221a0690e06c24162c6c04501031f2212963152f82800daa7